### PR TITLE
[CTS] Add more CTS coverage for urKernelSetArgSampler

### PR DESCRIPTION
**Pull request by omarahmed1111**
This PR adds more CTS tests to `urKernelSetArgSampler` entry-point by testing setting the kernel argument with different types of samplers.

### CODE REVIEW

File: test/conformance/kernel/urKernelSetArgSampler.cpp [hunk 1]
```diff
@@ -5,6 +5,65 @@
...
+                                    uur::SamplerCreateParamTy>::TearDown());
+    }
+
+    ur_sampler_handle_t sampler = nullptr;
+};
+
+UUR_TEST_SUITE_P(
+    urKernelSetArgSamplerTestWithParam,
+    ::testing::Combine(
+        ::testing::Values(true, false),
```
Comment (martygrant) [inline test/conformance/kernel/urKernelSetArgSampler.cpp:53]: I've seen in other test suites for parameterising tests with enabling/disabling of flags the use of `testing::ValuesIn(uur::BoolTestParam::makeBoolParam("UsePool")` which could maybe could be of use here and with the test name printing function.

File: test/conformance/kernel/urKernelSetArgSampler.cpp [hunk 2]
```diff
@@ -36,10 +95,13 @@ struct urKernelSetArgSamplerTest : uur::urBaseKernelTest {
...
 };
+
 UUR_INSTANTIATE_KERNEL_TEST_SUITE_P(urKernelSetArgSamplerTest);
 
-TEST_P(urKernelSetArgSamplerTest, Success) {
-    ASSERT_SUCCESS(urKernelSetArgSampler(kernel, 2, nullptr, sampler));
+TEST_P(urKernelSetArgSamplerTest, SuccessWithProps) {
+    ur_kernel_arg_sampler_properties_t props{
+        UR_STRUCTURE_TYPE_KERNEL_ARG_SAMPLER_PROPERTIES, nullptr};
+    ASSERT_SUCCESS(urKernelSetArgSampler(kernel, 2, &props, sampler));
```
Comment (martygrant) [inline test/conformance/kernel/urKernelSetArgSampler.cpp:104]: What's the meaning of the `2` here and could it be a named variable?

File: test/conformance/testing/include/uur/fixtures.h [hunk 3]
```diff
@@ -1068,6 +1068,14 @@ std::string deviceTestWithParamPrinter<BoolTestParam>(
     const ::testing::TestParamInfo<
         std::tuple<ur_device_handle_t, BoolTestParam>> &info);
 
+using SamplerCreateParamTy =
```
Comment (martygrant) [inline test/conformance/testing/include/uur/fixtures.h:1071]: Is the `Ty` at the end of this name a typo?

**Review by martygrant — APPROVED** [review]
LGTM, couple small queries.

File: test/conformance/kernel/urKernelSetArgSampler.cpp (hunk 1 above)
Comment (omarahmed1111) [reply test/conformance/kernel/urKernelSetArgSampler.cpp:53]: I think `makeBoolParam` was used mainly to generate meaningful names for tests that we use true and false for the param only like `usepool_true` / `usepool_false`. But in our case `normalized` and `unnormalized` name should be more clear I think, so I modified it to reflect that in the device printer.

File: test/conformance/kernel/urKernelSetArgSampler.cpp (hunk 2 above)
Comment (omarahmed1111) [reply test/conformance/kernel/urKernelSetArgSampler.cpp:105]: Thanks for pointing that, it should be the `arg_index` but I didn't notice that it is a magic number so changed that.

File: test/conformance/testing/include/uur/fixtures.h (hunk 3 above)
Comment (omarahmed1111) [reply test/conformance/testing/include/uur/fixtures.h:1071]: It should reflect an alias for a tuple that holds the sampler creation parameters. So, I guess `Ty` should reflect that it is like an alias type. But I could change that if you think that shouldn't be accurate.

File: test/conformance/kernel/urKernelSetArgSampler.cpp (hunk 1 above)
Comment (martygrant) [reply test/conformance/kernel/urKernelSetArgSampler.cpp:53]: Fair enough!

File: test/conformance/testing/include/uur/fixtures.h (hunk 3 above)
Comment (martygrant) [reply test/conformance/testing/include/uur/fixtures.h:1072]: Ah is this a naming convention? Never came across it before.

Comment (kbenzie) [reply test/conformance/testing/include/uur/fixtures.h:1072]: `Ty` is usually form `llvm::Type*` variables, though. I don't think its UR naming convention for an actual type. That would either be `T` in PascalCase names or `_t` in snake_case names.

Comment (omarahmed1111) [reply test/conformance/testing/include/uur/fixtures.h]: I changed it to a `T`. I didn't notice that `Ty` is not common in UR codebase.

**Review by kbenzie — APPROVED** [review]

**Pull request merged**